typescript/sdk/src/deploy/EvmModuleDeployer.ts

```diff
@@ -0,0 +1,288 @@
+      constructorArgs,
+      initializeArgs,
+    });
+  }
+
+  // Deploys the Implementation and Proxy for a given contract
+  public async deployProxiedContract<K extends keyof Factories>({
+    chain,
+    contractKey,
+    contractName,
+    proxyAdmin,
+    constructorArgs,
+    initializeArgs,
+  }: {
+    chain: ChainName;
+    contractKey: K;
+    contractName: string;
+    proxyAdmin: string;
+    constructorArgs: Parameters<Factories[K]['deploy']>;
+    initializeArgs?: Parameters<HyperlaneContracts<Factories>[K]['initialize']>;
+  }): Promise<HyperlaneContracts<Factories>[K]> {
+    // Try to initialize the implementation even though it may not be necessary
+    const implementation = await this.deployContractWithName({
+      chain,
+      contractKey,
+      contractName,
+      constructorArgs,
+      initializeArgs,
+    });
+
+    // Initialize the proxy the same way
+    return this.deployProxy({
+      chain,
+      implementation,
+      proxyAdmin,
+      initializeArgs,
+    });
+  }
+
+  // Deploys a proxy for a given implementation contract
+  protected async deployProxy<C extends ethers.Contract>({
+    chain,
+    implementation,
+    proxyAdmin,
+    initializeArgs,
+  }: {
+    chain: ChainName;
+    implementation: C;
+    proxyAdmin: string;
+    initializeArgs?: Parameters<C['initialize']>;
+  }): Promise<C> {
+    const isProxied = await isProxy(
+      this.multiProvider.getProvider(chain),
+      implementation.address,
+    );
+    if (isProxied) {
+      // if the implementation is already a proxy, do not deploy a new proxy
+      return implementation;
+    }
+
+    const constructorArgs = proxyConstructorArgs(
+      implementation,
+      proxyAdmin,
+      initializeArgs,
+    );
+    const proxy = await this.deployContractFromFactory({
+      chain,
+      factory: new TransparentUpgradeableProxy__factory(),
+      contractName: 'TransparentUpgradeableProxy',
+      constructorArgs,
+    });
+
+    return implementation.attach(proxy.address) as C;
+  }
+
+  // Adds verification artifacts to the verificationInputs map
+  protected addVerificationArtifacts({
+    chain,
+    artifacts,
+  }: {
+    chain: ChainName;
+    artifacts: ContractVerificationInput[];
+  }): void {
+    this.verificationInputs[chain] = this.verificationInputs[chain] || [];
+    artifacts.forEach((artifact) => {
+      this.verificationInputs[chain].push(artifact);
+    });
+  }
+
+  // Static deploy function used by Hook and ISM modules.
+  public static async deployStaticAddressSet({
+    chain,
+    factory,
+    values,
+    logger,
+    threshold = values.length,
+    multiProvider,
+  }: {
+    chain: ChainName;
+    factory: StaticThresholdAddressSetFactory | StaticAddressSetFactory;
+    values: Address[];
+    logger: Logger;
+    threshold?: number;
+    multiProvider: MultiProvider;
+  }): Promise<Address> {
+    const address = await factory['getAddress(address[],uint8)'](
+      values,
```

Comment:
should we `sort()` the values as according to https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/6620fe6367cc8cbc50d2edab4b057ccec1da41ba/solidity/contracts/libs/StaticAddressSetFactory.sol#L48

Reply:
yeah @yorhodes and i had a chat about this before - we ended up deciding that sorting should be something done by the caller/whoever writes the configuration in the first place - so that it simplifies the logic on the deployer/update/checking end of making sure it matches what we expect